android/src/main/java/io/radar/react/RNRadarModule.java

```diff
@@ -129,14 +177,21 @@ public boolean onRequestPermissionsResult(int requestCode, String[] permissions,
 
     @ReactMethod
     public void requestPermissions(boolean background, final Promise promise) {
-        PermissionAwareActivity activity = (PermissionAwareActivity)getCurrentActivity();
+        PermissionAwareActivity activity = (PermissionAwareActivity) getCurrentActivity();
         mPermissionsRequestPromise = promise;
         if (activity != null) {
-            if (Build.VERSION.SDK_INT >= 23) {
-                if (background && Build.VERSION.SDK_INT >= 29) {
-                    activity.requestPermissions(new String[] { Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_BACKGROUND_LOCATION }, PERMISSIONS_REQUEST_CODE, this);
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                if (background && Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                    activity.requestPermissions(new String[]{
+                            Manifest.permission.ACCESS_FINE_LOCATION,
+                            Manifest.permission.ACCESS_COARSE_LOCATION,
+                            Manifest.permission.ACCESS_BACKGROUND_LOCATION
```

Comment:
Been looking into permissions again, and this BACKGROUND_LOCATION permission needs to be included in a separate permission dialog than FINE/COARSE - Need to make this change before continuing with this PR.

Reply:
Fixed.